File: twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        vector<int> result(2);
        unordered_map<int, int> storeNum;
        int x = numbers.size();
        for(int i = 0;i < x;i ++){
            storeNum[numbers[i]] = i;
        }
        for(int i = 0;i < x;i ++){
            int gap = target - numbers[i];
            if(storeNum.find(gap) != storeNum.end() && storeNum[gap] != i){
                result[0]= i + 1;
                result[1]= storeNum[gap] + 1;
                break;
            }
        }
        return result;
    }
};

File: twoSum.py
class Solution:
    # @return a tuple, (index1, index2)
    def twoSum(self, num, target):
        di = {}
        count = 1
        for item in num:
            di[item]=count
            count+=1
        for x in num:
            gap = target - x
            if gap in di:
                a = num.index(x)+1
                b = di[gap]
                if a!=b:
                    return(a,b)

